File: cpp/algorithm/algorithm/21758.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

long long v[1000001], sum[1000001];

int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	int n;
	long long ans = 0;

	cin >> n;
	for (int i = 1; i <= n; i++) {
		cin >> v[i];
		sum[i] = v[i] + sum[i - 1];
	}

	for (int i = 2; i < n; i++) { // ²ÜÅë ¿À¸¥ÂÊ
		ans = max(ans, sum[n] - v[1] - v[i] + sum[n] - sum[i]);
	}

	for (int i = 2; i < n; i++) {
		ans = max(ans, sum[i - 1] + sum[n] - v[n] - v[i]);
	}

	for (int i = 2; i < n; i++) {
		ans = max(ans, sum[i] - v[1] + sum[n] - sum[i - 1] - v[n]);
	}

	cout << ans << endl;

	return 0;
}

File: python/21758.py
n = int(input())
a = list(map(int, input().split()))

ans = 0
s = []
s.append(a[0])

for i in range(1, n):
    s.append(s[i - 1] + a[i])

for i in range(1, n-1):
    ans = max(ans, s[n-2] + s[i-1] - a[i])

for i in range(1, n-1):
    ans = max(ans, s[n-1] - a[0]-a[i] +s[n-1] - s[i])

for i in range(1, n-1):
    ans = max(ans, s[n-2] - a[0] + a[i])

print(ans)